src/components/Filter/Filter.tsx

```diff
@@ -0,0 +1,52 @@
+import classNames from 'classnames';
+import { useState } from 'react';
+
+type Props = {
+  setFilter: (filter?: TypeFilter) => void
+};
+
+export enum TypeFilter {
+  ALL = 'all',
+  ACTIVE = 'active',
+  COMPLETED = 'completed',
+}
+
+export const Filter: React.FC<Props> = ({ setFilter }) => {
+  const [typeFilter, setTypeFilter] = useState(TypeFilter.ALL);
+  const filters = Object.values(TypeFilter);
+
+  const typeFunc = (type: TypeFilter) => {
+    switch (type) {
+      case TypeFilter.ALL:
+        setTypeFilter(type);
+        setFilter(TypeFilter.ALL);
```

Comment:
You can create useEffect that will do request for todos in case when filter change. 
Try to refactor it